src/Marten.Testing/Harness/IntegrationContext.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using Marten.Testing.Examples;
+using Xunit;
+
+namespace Marten.Testing.Harness
+{
+    [Collection("integration")]
+    public class IntegrationContext: StoreContext<DefaultStoreFixture>
+    {
+        private IDocumentSession _session;
+        private DocumentStore _store;
+
+        public IntegrationContext(DefaultStoreFixture fixture) : base(fixture)
+        {
+
+        }
+
+        /// <summary>
+        /// Customize the store configuration for one off tests.
+        /// The return value is the database schema
+        /// </summary>
+        /// <param name="configure"></param>
+        /// <returns></returns>
+        protected string StoreOptions(Action<StoreOptions> configure)
+        {
+            _session?.Dispose();
+            _session = null;
+
+            var options = new StoreOptions();
+            options.Connection(ConnectionSource.ConnectionString);
+
+            // Can be overridden
+            options.AutoCreateSchemaObjects = AutoCreate.All;
+            options.NameDataLength = 100;
+            options.DatabaseSchemaName = "special";
+
+            configure(options);
+
+            _store = new DocumentStore(options);
+
+            _store.Advanced.Clean.CompletelyRemoveAll();
+
+            return options.DatabaseSchemaName;
+        }
+
+        private bool _hasBuiltStore = false;
+
+        protected override DocumentStore theStore
+        {
+            get
+            {
+                if (_store != null) return _store;
+
+                if (!_hasBuiltStore)
```

Comment:
**CONCERN:** Is this code reachable? We're setting the _store in the constructor https://github.com/JasperFx/marten/pull/1463/files#diff-8f22a8190d9167240cb5140d183b18fbR39. `_store` is private and not exposed anywhere then it makes it readonly in fact.

Reply:
It is reachable, and it's getting around whether or not you're using the default store that's shared or using something overridden by StoreOptions